zuul-core/src/test/java/com/netflix/netty/common/proxyprotocol/OptionalHAProxyMessageDecoderTest.java

```diff
@@ -1,133 +0,0 @@
-
-        ByteBuf buf = Unpooled.wrappedBuffer(
-                "PROXY TCP4 192.168.0.1 124.123.111.111 10008 443\r\n".getBytes(StandardCharsets.US_ASCII));
-        channel.writeInbound(buf);
-
-        Object msg = channel.readInbound();
-        HAProxyMessage hapm = (HAProxyMessage) msg;
-
-        // The handler should remove itself.
-        assertNull(channel.pipeline().context(OptionalHAProxyMessageDecoder.NAME));
-
-        // TODO(carl-mastrangelo): make this always happen once netty has been upgraded.
-        if (hapm instanceof ReferenceCounted) {
-            hapm.release();
-        }
-    }
-
-    @Test
-    @Ignore // TODO(carl-mastrangelo): reenable this.
-    public void detectsSplitPpv1Message() {
-        OptionalHAProxyMessageDecoder decoder = new OptionalHAProxyMessageDecoder();
-        EmbeddedChannel channel = new EmbeddedChannel();
-        channel.pipeline().addLast(OptionalHAProxyMessageDecoder.NAME, decoder);
-
-        ByteBuf buf1 = Unpooled.wrappedBuffer(
-                "PROXY TCP4".getBytes(StandardCharsets.US_ASCII));
-        channel.writeInbound(buf1);
-        ByteBuf buf2 = Unpooled.wrappedBuffer(
-                "192.168.0.1 124.123.111.111 10008 443\r\n".getBytes(StandardCharsets.US_ASCII));
-        channel.writeInbound(buf2);
-
-        Object msg = channel.readInbound();
-        HAProxyMessage hapm = (HAProxyMessage) msg;
-
-        // The handler should remove itself.
-        assertNull(channel.pipeline().context(OptionalHAProxyMessageDecoder.NAME));
-
-        // TODO(carl-mastrangelo): make this always happen once netty has been upgraded.
-        if (hapm instanceof ReferenceCounted) {
-            hapm.release();
-        }
-    }
-
-    @Test
-    public void extraDataForwarded() {
-        OptionalHAProxyMessageDecoder decoder = new OptionalHAProxyMessageDecoder();
-        EmbeddedChannel channel = new EmbeddedChannel();
-        channel.pipeline().addLast(OptionalHAProxyMessageDecoder.NAME, decoder);
-
-        ByteBuf buf = Unpooled.wrappedBuffer(
-                "PROXY TCP4 192.168.0.1 124.123.111.111 10008 443\r\nPOTATO".getBytes(StandardCharsets.US_ASCII));
-        channel.writeInbound(buf);
-
-        Object msg = channel.readInbound();
-        HAProxyMessage hapm = (HAProxyMessage) msg;
-
-        // The handler should remove itself.
-        assertNull(channel.pipeline().context(OptionalHAProxyMessageDecoder.NAME));
-
-        // TODO(carl-mastrangelo): make this always happen once netty has been upgraded.
-        if (hapm instanceof ReferenceCounted) {
-            hapm.release();
-        }
-
-        Object msg2 = channel.readInbound();
-        ByteBuf readBuf = (ByteBuf) msg2;
-        assertEquals("POTATO", new String(ByteBufUtil.getBytes(readBuf), StandardCharsets.US_ASCII));
-        readBuf.release();
-    }
-
-    @Test
-    public void ignoresNonPpMessage() {
-        OptionalHAProxyMessageDecoder decoder = new OptionalHAProxyMessageDecoder();
```

Comment:
Same comment, this test case was useful.

Reply:
This is covered in the new test I added - `ElbProxyProtocolChannelHandlerTest::passThrough_ProxyProtocolEnabled_nonProxyBytes`